File: xNet/src/main/java/org/sqlunet/HasXId.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.sqlunet;

import androidx.annotation.Nullable;

/**
 * Has extended-id interface
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public interface HasXId
{
	/**
	 * Get extended class id (class=vn:class id,pb:rolesetid,fn:frame id)
	 *
	 * @return extended class id (class=vn:class,pb:rolesetid,fn:frame)
	 */
	@Nullable
	Long getXId();

	/**
	 * Get extended class id (class=vn:class id,pb:roleset id,fn:frame id)
	 *
	 * @return extended class id (class=vn:class id,pb:roleset id,fn:frame id)
	 */
	@Nullable
	Long getXClassId();

	/**
	 * Get extended member id (fn:lexunit)
	 *
	 * @return extended member id (fn:lexunit)
	 */
	@Nullable
	Long getXMemberId();

	/**
	 * Get extended sources
	 *
	 * @return extended sources
	 */
	@Nullable
	String getXSources();
}

File: xNet/src/main/java/org/sqlunet/HasXId.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet

/**
 * Has extended-id interface
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
interface HasXId {

    /**
     * Get extended class id (class=vn:class id,pb:rolesetid,fn:frame id)
     *
     * @return extended class id (class=vn:class,pb:rolesetid,fn:frame)
     */
    fun getXId(): Long?

    /**
     * Get extended class id (class=vn:class id,pb:roleset id,fn:frame id)
     *
     * @return extended class id (class=vn:class id,pb:roleset id,fn:frame id)
     */
    fun getXClassId(): Long?

    /**
     * Get extended member id (fn:lexunit)
     *
     * @return extended member id (fn:lexunit)
     */
    fun getXMemberId(): Long?

    /**
     * Get extended sources
     *
     * @return extended sources
     */
    fun getXSources(): String?
}
